torch/csrc/jit/script/class_type.cpp

```diff
@@ -6,6 +6,10 @@ namespace c10 {
 
 // This file exists because we need to reference module.h, which we can't from
 // c10. Sigh...
+FunctionType::FunctionType(Function* function)
+    : NamedType(TypeKind::FunctionType),
+      function_(function),
+      name_(function_->qualname()) {}
```

Comment:
Note that while this code is correct, it is susceptible to the order of fields in `FunctionType`. If for some reason the code will get shuffled in future, we might call `function_->qualname()` before `function_` is initialized. I think it would be safer if we use `name_(function->qualname())` instead of `name_(function_->qualname())`.